src/server_manager/web_app/app.ts

```diff
@@ -785,18 +806,21 @@ export class App {
     this.showMetricsOptInWhenNeeded(selectedServer, view);
 
     // Load "My Connection" and other access keys.
-    selectedServer.listAccessKeys()
-        .then((serverAccessKeys: server.AccessKey[]) => {
-          view.accessKeyRows = serverAccessKeys.map(this.convertToUiAccessKey.bind(this));
-          // Initialize help bubbles once the page has rendered.
-          setTimeout(() => {
-            view.initHelpBubbles();
-          }, 250);
-        })
-        .catch((error) => {
-          console.error(`Failed to load access keys: ${error}`);
-          this.appRoot.showError(this.appRoot.localize('error-keys-get'));
-        });
+    try {
+      const serverAccessKeys = await selectedServer.listAccessKeys();
+      view.accessKeyRows = serverAccessKeys.map(this.convertToUiAccessKey.bind(this));
+      if (!view.accessKeyDataLimit) {
+        view.accessKeyDataLimit = this.dataLimitToDisplayDataAmount(
+            await this.computeDefaultAccessKeyDataLimit(selectedServer, serverAccessKeys));
+      }
+      // Initialize help bubbles once the page has rendered.
+      setTimeout(() => {
+        view.initHelpBubbles();
```

Comment:
The "once" logic should be here, not in the UI component.

Reply:
Done in 689d8b9.